contracts/contracts/GrantRound.sol

```diff
@@ -0,0 +1,134 @@
+    _;
+  }
+
+  modifier roundEnd() {
+    require(block.timestamp >= endTime, "Method must be called after the active round has ended");
+    _;
+  }
+
+  /**
+   * @notice Instantiates a new grant round
+   * @param _owner Grant round owner that has permission to payout the matching pool
+   * @param _registry Address that contains the grant metadata
+   * @param _donationToken Address of the ERC20 token in which donations are made
+   * @param _startTime Unix timestamp of the start of the round
+   * @param _endTime Unix timestamp of the end of the round
+   * @param _metaPtr URL pointing to the grant round metadata
+   * @param _minContribution Miniumum donation amount that can be made using the given donation token
+   */
+  constructor(
+    address _owner,
+    address _registry,
+    IERC20 _donationToken,
+    uint256 _startTime,
+    uint256 _endTime,
+    string memory _metaPtr,
+    uint256 _minContribution
+  ) {
+    hasPaidOut = false;
+    owner = _owner;
+    registry = _registry;
+    donationToken = _donationToken;
+    startTime = _startTime;
+    endTime = _endTime;
+    metaPtr = _metaPtr;
+    minContribution = _minContribution;
+  }
+
+  /**
+   * @notice Before the round ends this method accepts matching pool funds
+   * @param _amount The amount of donation token that can be sent to the contract for the matching pool
+   */
+  function acceptMatchingPool(uint256 _amount) external beforeRoundEnd {
+    _safeTransferDonationToken(msg.sender, address(this), _amount);
+  }
+
+  /**
+   * @notice During an active round users can use this method to send donation tokens to a specified grant
+   * @param _donationAmount The number of tokens to be sent to grant receipient
+   * @param _grantId The id of the grant in the registry
+   */
+  function donateToGrant(uint256 _donationAmount, uint96 _grantId) external activeRound {
+    require(_donationAmount >= minContribution, "Donation amount must be greater than minimum contribution");
+
+    GrantRegistry gRegistry = GrantRegistry(registry);
+    require(gRegistry.isGrantInRegistry(_grantId), "Grant with given id does not exist in registry provided");
+
+    address payee = gRegistry.getGrantPayee(_grantId);
+    require(payee != address(0), "Payee not set in the grant metadata");
+    donationToken.transfer(payee, _donationAmount);
+
+    emit DonationSent(_grantId, address(donationToken), _donationAmount, payee, msg.sender);
+  }
+
+  /**
+   * @notice When the round ends the owner can send the remaining matching pool funds to a given address
+   * @param _payoutAddress An address to receive the remaining matching pool funds in the contract
+   */
+  function payoutGrant(address _payoutAddress) external roundEnd onlyOwner {
+    uint256 balance = donationToken.balanceOf(address(this));
+    donationToken.safeTransferFrom(address(this), _payoutAddress, balance);
```

Comment:
This could just be a regular `transfer` since we're transferring from this contract. No need for `transferFrom` here!

Reply:
This might should be called payoutGrants or approvePayout or something along those lines as it is transferring funds to somewhere else so that it can pay out multiple grants. I see this more as the approval to move funds to the destination responsible for paying out all of the grants.